core/internal/filetransfer/file_transfer_default.go

```diff
@@ -150,6 +190,112 @@ func (ft *DefaultFileTransfer) Download(task *DefaultDownloadTask) error {
 	return nil
 }
 
+// DownloadMultipart download the file using multiple http range requests.
+// You should always use [Download] instead of this method unless you are testing file write errorh handling.
+func (ft *DefaultFileTransfer) DownloadMultipart(task *DefaultDownloadTask, file WriterAt) error {
+	// Split file into multiple http range requests and distribute to multiple workers
+	requests := SplitHttpRanges(task.Size, ft.multipartDownloadPartSize)
+	numWorkers := min(runtime.NumCPU(), len(requests))
```

Comment:
Is downloading a CPU-bound task? Feels IO-bound to me. I don't think tying this to the number of CPUs is right; you'll just download unnecessarily slowly in Docker containers that limit CPUs, or make way too many parallel downloads on machines with a lot of CPUs available.